src/utils/metadata/parser.rs

```diff
@@ -174,8 +174,8 @@ impl Metadata for ParserMetadata {
         }
     }
 
-    fn get_trace(&self) -> Vec<PositionInfo> {
-        self.context.trace.clone()
+    fn get_token_at(&self, index: usize) -> Option<Token> {
```

Comment:
RustRover warns that `impl Metadata for ParserMetadata` functions are in a different order to `trait Metadata`, and offers to reorder them. I don't think it hurts to do this here.